internal/stores/sql.go

```diff
@@ -101,6 +101,11 @@ func NewSQLStore(conn gorm.Dialector, migrate bool, persistInterval time.Duratio
 		}
 	}
 
+	// Ensure the join table has an index on `db_host_id`.
+	if err := db.Exec("CREATE INDEX IF NOT EXISTS `idx_host_blocklist_entry_hosts` ON `host_blocklist_entry_hosts`(`db_host_id`)").Error; err != nil {
```

Comment:
`NOTE`: the alternative to running this query on startup, is to explicitly define a join table type and add all the appropriate `gorm` annotations. I like this better though the gorm annotations are awful with all the FK constraints and combined indices.